src/pl.rs

```diff
@@ -0,0 +1,98 @@
+//                              "effectiveDate":"2021-02-26",
+//                              "mid":3.7247}]}
+
+#[derive(Debug, Deserialize, Serialize)]
+struct NBPResponse<T> {
+    table: String,
+    currency: String,
+    code: String,
+    rates: Vec<T>,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+struct ExchangeRate {
+    no: String,
+    effectiveDate: String,
+    mid: f32,
+}
+
+pub struct PL {}
+
+impl etradeTaxReturnHelper::Residency for PL {
+    fn get_exchange_rate(&self, transaction_date: &str) -> Result<(String, f32), String> {
+        // proxies are taken from env vars: http_proxy and https_proxy
+        let http_proxy = std::env::var("http_proxy");
+        let https_proxy = std::env::var("https_proxy");
+
+        // If there is proxy then pick first URL
+        let base_client = ReqwestClient::builder();
+        let client = match &http_proxy {
+            Ok(proxy) => {
+                base_client.proxy(reqwest::Proxy::http(proxy).expect("Error setting HTTP proxy"))
+            }
+            Err(_) => base_client,
+        };
+        let client = match &https_proxy {
+            Ok(proxy) => {
+                client.proxy(reqwest::Proxy::https(proxy).expect("Error setting HTTP proxy"))
+            }
+            Err(_) => client,
+        };
+        let client = client.build().expect("Could not create REST API client");
+
+        let base_exchange_rate_url = "http://api.nbp.pl/api/exchangerates/rates/a/usd/";
+        let mut converted_date =
+            chrono::NaiveDate::parse_from_str(transaction_date, "%m/%d/%y").unwrap();
+
+        // Try to get exchange rate going backwards with dates till success
+        let mut is_success = false;
+        let mut exchange_rate = 0.0;
+        let mut exchange_rate_date: String = "N/A".to_string();
+        while is_success == false {
+            converted_date = converted_date
+                .checked_sub_signed(chrono::Duration::days(1))
+                .expect("Error traversing date");
+
+            let exchange_rate_url: String = base_exchange_rate_url.to_string()
+                + &format!("{}", converted_date.format("%Y-%m-%d"))
+                + "/?format=json";
+
+            let body = client.get(&(exchange_rate_url)).send();
+            let actual_body = body.expect(&format!(
+                "Getting Exchange Rate from NBP ({}) failed",
+                exchange_rate_url
+            ));
+            is_success = actual_body.status().is_success();
+            if is_success == true {
+                log::info!("RESPONSE {:#?}", actual_body);
+
+                let nbp_response = actual_body
+                    .json::<NBPResponse<ExchangeRate>>()
+                    .expect("Error converting response to JSON");
+                log::info!("body of exchange_rate = {:#?}", nbp_response);
+                exchange_rate = nbp_response.rates[0].mid;
+                exchange_rate_date = format!("{}", converted_date.format("%Y-%m-%d"));
```

Comment:
Maybe this converted_date could be converted to string once because this is happening second time here.